app/src/components/v-upload.vue

```diff
@@ -102,7 +105,7 @@ function useUpload() {
 
 				const uploadedFiles = await uploadFiles(Array.from(files), {
 					onProgressChange: (percentages) => {
-						progress.value = Math.round(
+						newUpload.progress.value = Math.round(
```

Comment:
Should the progress of the current upload component be shared to the store, or is that something we can just keep locally? 🤔

Reply:
On Google Drive, when you upload files, there's a small popup on right bottom corner that shows the status of the upload while you navigate through the app.

While we do not have this yet, I was thinking that all the upload logic could be moved into it's own logic so we can easily access the state for future improvements. Also this should be more consistent and clean.

I also understand that this will store a new number value on memory for each upload (single or multiple files) and I am ok with putting it as before, but in my perspective the memory used this way vs. the legibility and flexibility this gives to us should worth it. 🧐